pkg/commands/os.go

```diff
@@ -49,13 +49,52 @@ func NewOSCommand(log *logrus.Entry, config config.AppConfigurer) *OSCommand {
 // RunCommandWithOutput wrapper around commands returning their output and error
 func (c *OSCommand) RunCommandWithOutput(command string) (string, error) {
 	c.Log.WithField("command", command).Info("RunCommand")
-	splitCmd := str.ToArgv(command)
+	splitCmd := ToArgv(command)
 	c.Log.Info(splitCmd)
 	return sanitisedCommandOutput(
 		c.command(splitCmd[0], splitCmd[1:]...).CombinedOutput(),
 	)
 }
 
+// RunCommandWithOutputLive runs RunCommandWithOutputLiveWrapper
+func (c *OSCommand) RunCommandWithOutputLive(command string, output func(string) string) (errorMessage string, err error) {
+	return RunCommandWithOutputLiveWrapper(c, command, output)
+}
+
+// DetectUnamePass detect a username / password question in a command
+// ask is a function that gets executen when this function detect you need to fillin a password
+// The ask argument will be "username" or "password" and expects the user's password or username back
+func (c *OSCommand) DetectUnamePass(command string, ask func(string) string) error {
+	ttyText := ""
+	errMessage, err := c.RunCommandWithOutputLive(command, func(word string) string {
+		ttyText = ttyText + " " + word
+
+		// detect username question
+		detectUname, _ := regexp.MatchString(`Username\s*for\s*'.+':`, ttyText)
+		if detectUname {
+			// reset the text and return the user's username
+			ttyText = ""
+			return ask("username")
+		}
+
+		// detect password question
+		detectPass, _ := regexp.MatchString(`Password\s*for\s*'.+':`, ttyText)
```

Comment:
there is some code duplication here. Perhaps you could define a map like so:
```
	prompts := map[string]string{
		"password": `Password\s*for\s*'.+':`,
		"username": `Username\s*for\s*'.+':`,
	}

	for prompt, pattern := range prompts {
		detectUname, _ := regexp.MatchString(pattern, ttyText)
		if detectUname {
			// reset the text and return the user's username
			ttyText = ""
			return ask(prompt)
		}
	}
```

Reply:
Thx I'll do that